### PR TITLE
docs: Documentation update for release 0.2.0

### PR DESCRIPTION
**Pull request by onel**
# What does this PR do ?

Small docs update for [0.2.0](https://github.com/NVIDIA-NeMo/Automodel/releases/tag/v0.2.0) release.

# Changelog

- add refence to gpt oss 
- describe checkpoint symbolic Links

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: docs/guides/checkpointing.md
Comment (jgerh) [inline docs/guides/checkpointing.md]: suggest using language-tagged fenced code block for syntax highlighting

```suggestion
```yaml
```

File: docs/guides/checkpointing.md [hunk 1]
```diff
@@ -33,6 +33,14 @@ checkpoint:
 
 > **Note:** The optimizer states are _always_ saved in DCP (`.distcp` extension) format.
```
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
::: {note}
The optimizer states are _always_ saved in DCP (`.distcp` extension) format.
:::
```

File: docs/guides/checkpointing.md [hunk 2]
```diff
@@ -33,6 +33,14 @@ checkpoint:
...
+
+NeMo AutoModel automatically creates symbolic links in the checkpoint directory to provide convenient access to important checkpoints:
+
+- **LATEST**: Points to the most recently saved checkpoint. This is useful for resuming training from the last saved state.
+- **LOWEST_VAL**: Points to the checkpoint with the lowest validation score/loss. This provides easy access to the best-performing checkpoint based on validation metrics, making it ideal for model evaluation or deployment.
+
+These symbolic links eliminate the need to manually track checkpoint names or search through directories to find the best model. When validation is enabled in your training run, both links are automatically maintained and updated as training progresses.
 
 ## Safetensors
 To ensure seamless integration with the Hugging Face ecosystem, NeMo Automodel saves checkpoints in the [Safetensors](https://github.com/huggingface/safetensors) format. Safetensors is a memory-safe, zero-copy alternative to Python's pickle (Pytorch .bin), natively supported by Hugging Face Transformers, offering both safety and performance advantages over Python pickle-based approaches.
```
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
To ensure seamless integration with the Hugging Face ecosystem, NeMo Automodel saves checkpoints in the [Safetensors](https://github.com/huggingface/safetensors) format. Safetensors is a memory-safe, zero-copy alternative to Python's pickle (PyTorch .bin), natively supported by Hugging Face Transformers, offering both safety and performance advantages over Python pickle-based approaches.
```

File: docs/guides/checkpointing.md
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
```bash
```

File: docs/guides/checkpointing.md [hunk 3]
```diff
@@ -67,6 +75,7 @@ The `checkpoints/` should have the following contents:
 ```
 checkpoints/
 ├── LATEST -> epoch_0_step_19
+├── LOWEST_VAL -> epoch_0_step_19
 └── epoch_0_step_19
```
Comment (jgerh) [inline docs/guides/checkpointing.md]: Please verify this suggested change.


```suggestion
├── LATEST -> epoch_0_step_20
├── LOWEST_VAL -> epoch_0_step_20
└── epoch_0_step_20
```

File: docs/guides/checkpointing.md
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
```bash
```

File: docs/guides/checkpointing.md [hunk 4]
```diff
@@ -130,6 +139,7 @@ After training, you'll get a compact, consolidated Safetensors checkpoint that c
 ```
 checkpoints/
 ├── LATEST -> epoch_0_step_19
+├── LOWEST_VAL -> epoch_0_step_19
 ├── epoch_0_step_19
```
Comment (jgerh) [inline docs/guides/checkpointing.md]: Please verify suggested change

```suggestion
├── LATEST -> epoch_0_step_20
├── LOWEST_VAL -> epoch_0_step_20
├── epoch_0_step_20
```

File: docs/guides/checkpointing.md
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
The example below showcases the direct compatibility of NeMo Automodel with Hugging Face and PEFT:
```

Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
NeMo Automodel also offers native PyTorch DCP checkpointing support (`.distcp` extension). Similar to Safetensors, it also provides the same features of load-time resharding and parallel saving.
```

File: docs/guides/checkpointing.md [hunk 5]
```diff
@@ -189,6 +199,7 @@ After 20 steps, the following checkpoint will be saved:
 ```
 checkpoints/
 ├── LATEST -> epoch_0_step_19
+├── LOWEST_VAL -> epoch_0_step_19
 └── epoch_0_step_19
```
Comment (jgerh) [inline docs/guides/checkpointing.md]: Please verify suggested change


```suggestion
├── LATEST -> epoch_0_step_20
├── LOWEST_VAL -> epoch_0_step_20
└── epoch_0_step_20
```

File: docs/guides/checkpointing.md
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
If you rerun the script, NeMo Automodel automatically detects and restores the most recent checkpoint.
```

File: docs/guides/checkpointing.md [hunk 6]
```diff
@@ -215,7 +226,7 @@ uv run torchrun --nproc-per-node=2 examples/llm_finetune/finetune.py --step_sche
 
 ## Asynchronous Checkpointing
 
-NeMo AutoModel can write checkpoints asynchronously to reduce training stalls caused by I/O. When enabled, checkpoint writes are scheduled in the background using PyTorch Distributed Checkpointing’s async API while training continues.
+NeMo AutoModel can write checkpoints asynchronously to reduce training stalls caused by I/O. When enabled, checkpoint writes are scheduled in the background using PyTorch Distributed Checkpointing's async API while training continues.
```
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
NeMo Automodel can write checkpoints asynchronously to reduce training stalls caused by I/O. When enabled, checkpoint writes are scheduled in the background using PyTorch Distributed Checkpointing's async API while training continues.
```

File: docs/guides/checkpointing.md [hunk 7]
```diff
@@ -215,7 +226,7 @@ uv run torchrun --nproc-per-node=2 examples/llm_finetune/finetune.py --step_sche
...
 
 - **Enable** (YAML):
   ```
```
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
  ```yaml
```

File: docs/guides/checkpointing.md
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
You can also save additional states in NeMo Automodel. By default, we also automatically checkpoint the `dataloader`, `rng`, and `step_scheduler` states which are necessary to resume training accurately. In full, a Safetensors consolidated checkpoint will look like this:
```

File: docs/guides/checkpointing.md [hunk 8]
```diff
@@ -232,6 +243,7 @@ You can also save additional states in NeMo AutoModel. By default, we also autom
 ```
 checkpoints/
 ├── LATEST -> epoch_0_step_19
+├── LOWEST_VAL -> epoch_0_step_19
 ├── epoch_0_step_19
```
Comment (jgerh) [inline docs/guides/checkpointing.md]: Please verify suggested change


```suggestion
├── LATEST -> epoch_0_step_20
├── LOWEST_VAL -> epoch_0_step_20
├── epoch_0_step_20
```

File: docs/guides/checkpointing.md
Comment (jgerh) [inline docs/guides/checkpointing.md]: AI is flagging this line, please verify: Without this parameter, calling self.new_state.state_dict() would have resulted in a TypeError, as Python automatically passes the instance itself as the first argument to the method. I applied the same fix to load_state_dict on line 298.

```suggestion
    def state_dict(self) -> dict[str, Any]:
```

Comment (jgerh) [inline docs/guides/checkpointing.md]: Agent flagged this line, please check.

```suggestion
    def load_state_dict(self, state_dict: dict[str, Any]) -> None:
```

Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
# Checkpointing in NeMo Automodel
```

Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
NeMo Automodel checkpoints capture the complete state of a distributed training run across multiple GPUs or nodes. This reduces memory overhead, improves GPU utilization, and allows training to be resumed with a different parallelism strategy.
```

Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
NeMo Automodel writes checkpoints in two formats: [Hugging Face Safetensors](https://github.com/huggingface/safetensors) and [PyTorch Distributed Checkpointing (DCP)](https://docs.pytorch.org/docs/stable/distributed.checkpoint.html). It also supports two layouts:
```

File: docs/guides/checkpointing.md [hunk 9]
```diff
@@ -33,6 +33,14 @@ checkpoint:
 
 > **Note:** The optimizer states are _always_ saved in DCP (`.distcp` extension) format.
 
+## Checkpoint Symbolic Links
+
+NeMo AutoModel automatically creates symbolic links in the checkpoint directory to provide convenient access to important checkpoints:
```
Comment (jgerh) [inline docs/guides/checkpointing.md]:
```suggestion
NeMo Automodel automatically creates symbolic links in the checkpoint directory to provide convenient access to important checkpoints:
```

File: docs/model-coverage/llm.md
Comment (jgerh) [inline docs/model-coverage/llm.md]:
```suggestion
NeMo Automodel supports the [AutoModelForCausalLM](https://huggingface.co/transformers/v3.5.1/model_doc/auto.html#automodelforcausallm) in the [Text Generation](https://huggingface.co/models?pipeline_tag=text-generation&sort=trending) category. During preprocessing, it uses `transformers.AutoTokenizer`, which is sufficient for most LLM cases. If your model requires custom text handling, such as for reasoning tasks, you can override the default tokenizer during the data preparation stage.
```

Comment (jgerh) [inline docs/model-coverage/llm.md]:
```suggestion
Please see our [Fine-Tuning Guide](../guides/llm/finetune.md) to learn how you can apply both of these fine-tuning methods to your data.
```

Comment (jgerh) [inline docs/model-coverage/llm.md]:
```suggestion
In these guides, we use the `SQuAD v1.1` dataset for demonstration purposes, but you can specify your own data as needed.
```

Comment (jgerh) [inline docs/model-coverage/llm.md]:
```suggestion
1. Choose your approach (PEFT or SFT). See our [Fine-Tuning Guide](../guides/llm/finetune.md).
```

Comment (jgerh) [inline docs/model-coverage/llm.md]: best practice to use consistent bullet point syntax (hyphen)

```suggestion
   - Model and dataset configuration
   - Training recipe setup
   - Inference with fine-tuned models
   - Model sharing via Hugging Face Hub
   - Deployment with vLLM
```

Comment (jgerh) [inline docs/model-coverage/llm.md]:
```suggestion
  - Customize training parameters
  - Monitor progress
  - Save and share checkpoints
  - Deploy the fine-tuned model with optimized inference
```

**Comment by onel** [comment]
Applied the suggestions. Thanks for the review

File: docs/model-coverage/llm.md [hunk 10]
```diff
@@ -15,7 +15,7 @@ To run LLMs with NeMo Automodel, make sure you're using NeMo container version `
 For other installation options (e.g., uv), please see our [Installation Guide](../guides/installation.md).
 
 ## Supported Models
-NeMo Automodel supports the <a href=https://huggingface.co/transformers/v3.5.1/model_doc/auto.html#automodelforcausallm>AutoModelForCausalLM<a> in the <a href="https://huggingface.co/models?pipeline_tag=text-generation&sort=trending">Text Generation<a> category. During preprocessing, it uses `transformers.AutoTokenizer`, which is sufficient for most LLM cases. If your model requires custom text handling, such as for reasoning tasks, you can override the default tokenizer during the data preparation stage.
+NeMo Automodel supports the [AutoModelForCausalLM](https://huggingface.co/transformers/v3.5.1/model_doc/auto.html#automodelforcausallm) in the [Text Generation](https://huggingface.co/models?pipeline_tag=text-generation&sort=trending) category. During preprocessing, it uses `transformers.AutoTokenizer`, which is sufficient for most LLM cases. If your model requires custom text handling, such as for reasoning tasks, you can override the default tokenizer during the data preparation stage.
```
Comment (adil-a) [inline docs/model-coverage/llm.md:18]: Why do we need to change this? Seems like it renders fine on https://docs.nvidia.com/nemo/automodel/latest/model-coverage/llm.html#supported-models

Comment (jgerh) [reply docs/model-coverage/llm.md:18]: These are two link styles, HTML Style and standard Markdown. Since this is a Sphinx project with MyST parser, we use standard markdown syntax.

**Comment by onel** [comment]
Glad to hear @adil-a. I'm planning on contributing more. Any parts that you think need more attention?
